core/src/main/scala/org/http4s/multipart/Boundary.scala

```diff
@@ -29,27 +30,43 @@ final case class Boundary(value: String) extends AnyVal {
 }
 
 object Boundary {
-  private val BoundaryLength = 40
+  private val BoundaryLength = 41
   val CRLF = "\r\n"
 
-  private val DIGIT = ('0' to '9').toList
-  private val ALPHA = ('a' to 'z').toList ++ ('A' to 'Z').toList
-  // ' ' and '?' are also allowed by spec, but mean we need to quote
-  // the boundary in the media type, which causes some implementations
-  // pain.
-  private val OTHER = """'()+_,-./:=""".toSeq
-  private val CHARS = DIGIT ++ ALPHA ++ OTHER
-  private val nchars = CHARS.length
-  private val rand = new Random()
-
-  private def nextChar = CHARS(rand.nextInt(nchars - 1))
-  private def stream: CollectionCompat.LazyList[Char] =
-    CollectionCompat.LazyList.continually(nextChar)
-  // Don't use filterNot it works for 2.11.4 and nothing else, it will hang.
-  private def endChar: Char = stream.find(_ != ' ').getOrElse('X')
-  private def value(l: Int): String = stream.take(l).mkString
-
-  def create: Boundary = Boundary(value(BoundaryLength) + endChar)
+  private val alphabet = {
+    val arr = Array.newBuilder[Char]
+    arr ++= ('0' to '9')
+    arr ++= ('a' to 'z')
+    arr ++= ('A' to 'Z')
+    // Many more chars are allowed (see bchars definition in
+    // https://www.rfc-editor.org/rfc/rfc2046), but these are known to
+    // be robust in implementation.  Specifically, they don't trigger
+    // quotation in the Content-Type parameter.
+    arr += '-'
+    arr += '_'
+    arr.result()
+  }
+  private val nchars = alphabet.length
+  private val defaultRandom = new Random()
+
+  private def nextChar(random: Random) = alphabet(random.nextInt(nchars - 1))
+  private def stream(random: Random): CollectionCompat.LazyList[Char] =
+    CollectionCompat.LazyList.continually(nextChar(random))
+  private def value(random: Random, l: Int): String =
+    stream(random).take(l).mkString
+
+  @deprecated("Impure. Use fromScalaRandom", "0.22.14")
+  def create: Boundary = unsafeCreate()
+
+  /** Create a new MIME boundary. */
+  def fromScalaRandom[F[_]](random: Random)(implicit F: Sync[F]): F[Boundary] =
```

Comment:
I started in 0.22 because this is arguably a bug.  In 0.23, this would better take a Cats-Effect `Random`.

Reply:
[Moved to `Multiparts`]